test/tbb/test_partitioner.cpp

```diff
@@ -138,3 +140,53 @@ TEST_CASE("Threads respect task affinity") {
     task_affinity_retention::relaxed_test();
     task_affinity_retention::strict_test();
 }
+
+template <typename Range>
+void test_custom_range() {
+    int num_trials = 100;
+
+    std::vector<std::vector<std::size_t>> results(num_trials);
+    oneapi::tbb::mutex results_mutex;
+
+    for (int i = 0; i < num_trials; ++i) {
+        oneapi::tbb::parallel_for(Range(0, int(100 * utils::get_platform_max_threads()), 1), [&] (const Range& r) {
+            oneapi::tbb::mutex::scoped_lock lock(results_mutex);
+            results[i].push_back(r.size());
+        }, oneapi::tbb::static_partitioner{});
+    }
+
+    for (auto& res : results) {
+        REQUIRE(res.size() == utils::get_platform_max_threads());
+
+        std::size_t min_size = *std::min_element(res.begin(), res.end());
+        for (auto elem : res) {
+            REQUIRE(min_size * 2 + 1 >= elem);
```

Comment:
I think for the static partitioner it would be good to add test that checks tasks distribution with a known tolerance. And, for the ranges that support proportional split, to test that task sizes do not differ on more than 1 or maybe 2 iterations.

Reply:
What do you mean by "that checks tasks distribution with a known tolerance" ?

Reply:
Well, I meant to have the test that checks task distribution and execution more accurately for a given partitioner. For example, for the `static_partitioner` we know that all subranges are of the same size, except for one subrange that should not differ by one iteration. We also know that ideally these subranges should be executed by different threads and we also know in what circumstances that ideal execution might not happen so we can detect such situations in the test and still make it pass, and then, perhaps, check that non-ideal execution does not happen too often. This is what I wanted to have a test for - to detect incorrect "range <=> partitioner <=> execution engine"-relationship as early as possible in the SDL.

I think we need to take back (some of the) tests from [the classic TBB partitioner testing](https://github.com/oneapi-src/oneTBB/blob/v2020.3/src/test/test_partitioner_whitebox.cpp).

Reply:
Are you agree that for the proposed change the proposed test is enough? And we could create separate issue to add such testing for partitioners.

Reply:
Yes, I agree.